app/scaffold-controllers/drawerController.js

```diff
@@ -20,57 +22,45 @@ function(
     var DrawerController = function(drawer, leftMenu) {
         this.drawer = drawer;
         this.leftMenu = leftMenu;
-        this.activeItemId = null;
-
-        this._bindEvents();
     };
 
     var initLeftMenu = function(drawer, leftMenu) {
-        leftMenu.navigate('file:///scaffold-www/left-menu.html');
+        Astro.registerRpcMethod(AppRpc.names.menuItems, [], function(res) {
+            res.send(null, MenuConfig.menuItems);
+        });
+
+        leftMenu.navigate('file:///scaffold-www/html/left-drawer.html');
+        leftMenu.disableScrolling();
         drawer.setLeftMenu(leftMenu);
 
         return drawer;
     };
 
-    var initNavigationItems = function(drawer, tabItems, counterBadgeController, cartEventHandler, errorController) {
+    var initNavigationItems = function(drawer, counterBadgeController, cartEventHandler, errorController) {
         var drawerEventHandler = function() {
             drawer.showLeftMenu();
         };
 
-        drawer.itemViews = {};
-        drawer.itemControllers = {};
-
-        // Make sure all tabViews are set up
-        return Promise.all(tabItems.map(function(tab) {
-            //Init a new NavigationController
-            var navigationControllerPromise = NavigationController.init(
-                tab,
+        // The navigationController requires an id. Since we are using a drawer
+        // layout, we have 1 main navigation plugin -- thus its ID is set to 1.
+        var controllerID = 1;
+        return NavigationController.init(
+                controllerID,
+                BaseConfig.baseURL,
                 counterBadgeController,
                 cartEventHandler,
                 errorController,
                 drawerEventHandler
-            );
-
-            return navigationControllerPromise.then(function(navigationController) {
-                    drawer.itemControllers[tab.id] = navigationController;
-                    drawer.itemViews[tab.id] = navigationController.viewPlugin;
-
-                    return drawer;
-                });
-
-            })).then(function() {
+            ).then(function(navigationController) {
+                drawer.navigationController = navigationController;
```

Comment:
Now that we don't have to manage a stack of `navigationController`s I think it would be more clear if we had the navigationController as a member of the `drawerController`. I think it is possible to return the `navigationController` from `initNavigationItems` instead of returning the `drawer` and then pass in the `navigationController` to the constructor for the `drawerController`. What do you think?